diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..549560b
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,44 @@
+# CMakeLists.txt for lazyusf
+project(lazyusf)
+
+set(LAZYUSF_SOURCES
+    audio.c
+    cpu.c
+    dma.c
+    exception.c
+    interpreter_cpu.c
+    interpreter_ops.c
+    main.c
+    memory.c
+    pif.c
+    registers.c
+    tlb.c
+    usf.c
+    rsp/rsp.c
+    rsp_hle/alist.c
+    rsp_hle/alist_audio.c
+    rsp_hle/alist_naudio.c
+    rsp_hle/alist_nead.c
+    rsp_hle/audio.c
+    rsp_hle/cicx105.c
+    rsp_hle/hle.c
+    rsp_hle/jpeg.c
+    rsp_hle/memory.c
+    rsp_hle/mp3.c
+    rsp_hle/musyx.c
+    rsp_hle/plugin.c
+)
+
+add_library(lazyusf STATIC ${LAZYUSF_SOURCES})
+
+target_compile_definitions(lazyusf PRIVATE
+    PSS_STYLE=1
+    LSB_FIRST
+    __LIBUSF_INTERNAL__
+)
+
+target_include_directories(lazyusf PUBLIC 
+    . 
+    rsp 
+    rsp_hle
+)
