diff --git a/player/vgmplayer.hpp b/player/vgmplayer.hpp
--- a/player/vgmplayer.hpp
+++ b/player/vgmplayer.hpp
@@ -157,6 +157,7 @@
 	UINT8 GetDeviceOptions(UINT32 id, PLR_DEV_OPTS& devOpts) const;
 	UINT8 SetDeviceMuting(UINT32 id, const PLR_MUTE_OPTS& muteOpts);
 	UINT8 GetDeviceMuting(UINT32 id, PLR_MUTE_OPTS& muteOpts) const;
+	UINT8 SetDeviceVolume(UINT32 id, UINT16 volume);
 	// player-specific options
 	UINT8 SetPlayerOptions(const VGM_PLAY_OPTIONS& playOpts);
 	UINT8 GetPlayerOptions(VGM_PLAY_OPTIONS& playOpts) const;
diff --git a/player/vgmplayer.cpp b/player/vgmplayer.cpp
--- a/player/vgmplayer.cpp
+++ b/player/vgmplayer.cpp
@@ -787,6 +787,26 @@
 		return 0x80;	// bad device ID
 	
 	muteOpts = _devOpts[optID].muteOpts;
+	return 0x00;
+}
+
+UINT8 VGMPlayer::SetDeviceVolume(UINT32 id, UINT16 volume)
+{
+	size_t optID = DeviceID2OptionID(id);
+	if (optID == (size_t)-1)
+		return 0x80;	// bad device ID
+
+	size_t devID = _optDevMap[optID];
+	if (devID < _devices.size())
+	{
+		CHIP_DEVICE& chipDev = _devices[devID];
+		VGM_BASEDEV* clDev;
+		for (clDev = &chipDev.base; clDev != NULL; clDev = clDev->linkDev)
+		{
+			clDev->resmpl.volumeL = volume;
+			clDev->resmpl.volumeR = volume;
+		}
+	}
 	return 0x00;
 }
 
